From: Jeremy Bicha <jbicha@debian.org>
Date: Thu, 10 Mar 2022 07:50:04 -0500
Subject: Revert "status/bluetooth: Clear list of connected signals on adapter
 changes"

This reverts commit eeabdd150c70ff003a5cca60039ec8f2c4e5dbd0.

Ubuntu 22.04 LTS isn't using the new gnome-bluetooth series
---
 js/ui/status/bluetooth.js | 15 ++++-----------
 1 file changed, 4 insertions(+), 11 deletions(-)

diff --git a/js/ui/status/bluetooth.js b/js/ui/status/bluetooth.js
index 5e985dd..187e106 100644
--- a/js/ui/status/bluetooth.js
+++ b/js/ui/status/bluetooth.js
@@ -27,17 +27,6 @@ class Indicator extends PanelMenu.SystemIndicator {
         this._hadSetupDevices = global.settings.get_boolean(HAD_BLUETOOTH_DEVICES_SETUP);
 
         this._client = new GnomeBluetooth.Client();
-        this._client.connect('notify::default-adapter', () => {
-            const newAdapter = this._client.default_adapter ?? null;
-
-            if (newAdapter && this._adapter)
-                this._setHadSetupDevices(this._getDeviceInfos().length > 0);
-
-            this._adapter = newAdapter;
-
-            this._deviceNotifyConnected.clear();
-            this._sync();
-        });
         this._client.connect('notify::default-adapter-powered', this._sync.bind(this));
 
         this._proxy = new RfkillManagerProxy(Gio.DBus.session, BUS_NAME, OBJECT_PATH,
@@ -144,6 +133,10 @@ class Indicator extends PanelMenu.SystemIndicator {
         const connectedDevices = devices.filter(dev => dev.connected);
         const nConnectedDevices = connectedDevices.length;
 
+        if (this._client.default_adapter && this._adapter)
+            this._setHadSetupDevices(devices.length > 0);
+        this._adapter = this._client.default_adapter ?? null;
+
         let sensitive = !Main.sessionMode.isLocked && !Main.sessionMode.isGreeter;
 
         this.menu.setSensitive(sensitive);
